### PR TITLE
xmlcopyeditor: init at 1.2.1.3

### PR DESCRIPTION
**Pull request by candeira**
###### Motivation for this change

I use xmlcopyeditor, and it's not present in nixpkgs

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/xmlcopyeditor/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchurl {
+    name = "${name}.tar.gz";
+    url = "https://sourceforge.net/projects/xml-copy-editor/files/xmlcopyeditor-linux/${version}/${name}.tar.gz/download";
+    sha256 = "0bwxn89600jbrkvlwyawgc0c0qqxpl453mbgcb9qbbxl8984ns4v";
+  };
+
+  buildInputs = [ aspell boost expat expect intltool libxml libxslt pcre xercesc wxwidgets ];
+
+  C_INCLUDE_PATH = "${pkgs.libxml2.dev}/include/libxml2";
```
Comment (Infinisil) [inline pkgs/applications/editors/xmlcopyeditor/default.nix]: This attribute doesn't seem to be used so it should be removed.

File: pkgs/applications/editors/xmlcopyeditor/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  CPLUS_INCLUDE_PATH = "${pkgs.libxml2.dev}/include/libxml2";
```
Comment (Infinisil) [inline pkgs/applications/editors/xmlcopyeditor/default.nix]: Add libxml2 to the argument list at the top instead.

File: pkgs/applications/editors/xmlcopyeditor/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchurl, pkgs, aspell, boost, expat, expect, intltool, libxml, libxslt, pcre, xercesc, wxwidgets }:
+
+stdenv.mkDerivation rec {
+  name = "xmlcopyeditor-${version}";
+  version = "1.2.1.3";
+
+  src = fetchurl {
+    name = "${name}.tar.gz";
+    url = "https://sourceforge.net/projects/xml-copy-editor/files/xmlcopyeditor-linux/${version}/${name}.tar.gz/download";
```
Comment (Infinisil) [inline pkgs/applications/editors/xmlcopyeditor/default.nix]: Use the mirror url instead: `mirror://sourceforge/project/xml-copy-editor/xmlcopyeditor-linux/${version}/${name}.tar.gz`. Then the name will also be set to the last part automatically, so you can remove your name attribute.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -18690,6 +18690,11 @@ with pkgs;
 
   xmacro = callPackage ../tools/X11/xmacro { };
 
+  xmlcopyeditor = callPackage ../applications/editors/xmlcopyeditor {
+    libxml = libxml2;
+    wxwidgets = wxGTK;
```
Comment (Infinisil) [inline pkgs/top-level/all-packages.nix]: Change the arguments `libxml`/`wxwidgets` to `libxml2`/`wxGTK` at the top of `default.nix` instead.

**Review by infinisil — CHANGES_REQUESTED** [review]
When I run it I get the error
```
SSE2 is enabled in Xerces-C++ library. Xerces-C++ didn't use them in a thread-safe way. It may cause program crashes (segmentation faults).

If it happens, please try compiling Xerces-C++ with SSE2 disabled.
```

Can you try do that? This would involve adding support for disabling this SSE2 thing in xerces and usint `xercesc.override { enableSSE2 = false; }` in your build inputs instead.

Other than this error the editor is working just fine, nice first PR!

File: pkgs/applications/editors/xmlcopyeditor/default.nix (hunk 3 above)
Comment (jtojnar) [reply pkgs/applications/editors/xmlcopyeditor/default.nix]: I think you cand do even shorter: `mirror://sourceforge/xml-copy-editor/${name}.tar.gz`

Comment (Infinisil) [reply pkgs/applications/editors/xmlcopyeditor/default.nix]: It does indeed!

**Review by matthewbauer — CHANGES_REQUESTED** [review]
See @Infinisil's review

**Comment by jtojnar** [comment]
Any progress on this?

**Comment by c0bw3b** [comment]
@GrahamcOfBorg build xmlcopyeditor

File: pkgs/applications/editors/xmlcopyeditor/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, aspell, boost, expat, expect, intltool, libxml2, libxslt, pcre, wxGTK, xercesc }:
+
+stdenv.mkDerivation rec {
+  name = "xmlcopyeditor-${version}";
+  version = "1.2.1.3";
```
Comment (jtojnar) [inline pkgs/applications/editors/xmlcopyeditor/default.nix]: 1.2.1.5 is available

Comment (c0bw3b) [reply pkgs/applications/editors/xmlcopyeditor/default.nix]: It's a WIP macOS (only?) port as I understand it.

http://xml-copy-editor.sourceforge.net/ and https://sourceforge.net/projects/xml-copy-editor/files/xmlcopyeditor-linux/ still points to v1.2.1.3 for Linux (or Windows).

Comment (c0bw3b) [reply pkgs/applications/editors/xmlcopyeditor/default.nix]: The second link is not the front website but the list of source archives for Linux.
https://repology.org/metapackage/xmlcopyeditor/versions
No one ships v1.2.1.5.

So I'm going to stick to the version intended by the original author of this PR who's going to be its maintainer. @candeira should open a new PR later once upstream got its release process straighten out.

**Pull request merged**